src/VisualStudio/IntegrationTest/TestUtilities/InProcess/VisualStudio_InProc_Telemetry.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.Telemetry;
+
+namespace Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess
+{
+    internal partial class VisualStudio_InProc : InProcComponent
+    {
+        public void EnableTestTelemetryChannel()
+        {
+            InvokeOnUIThread(() =>
+            {
+                // detach if there is one already attached
+                TelemetryService.DetachTestChannel(LoggerTestChannel.Instance);
+
+                // make sure we clear any remaining events before attach it back
+                LoggerTestChannel.Instance.Clear();
+
+                // attach logger channel to the telemetry session
+                TelemetryService.AttachTestChannel(LoggerTestChannel.Instance);
+            });
+        }
+
+        public void DisableTestTelemetryChannel()
+        {
+            InvokeOnUIThread(() =>
+            {
+                // detach if there is one attached
+                TelemetryService.DetachTestChannel(LoggerTestChannel.Instance);
+
+                // make sure we clear any leftout events before attach it back
+                LoggerTestChannel.Instance.Clear();
+            });
+        }
+
+        public string[] GetTelemetryEventNames()
+            => LoggerTestChannel.Instance.GetEvents().Select(e => e.Name).ToArray();
+
+        /// <summary>
+        /// Logger channel collects events and provide them as a list, starting from the beginning.
+        /// One of the functionality is to clear current list of the events.
+        /// </summary>
+        private sealed class LoggerTestChannel : ITelemetryTestChannel
```

Comment:
I'm just going to remove the comment. The implementation has now diverged enough that I don't think it matters.